systemd/systemd.go

```diff
@@ -613,14 +623,37 @@ func (s *systemd) getUnitStatus(properties []string, unitNames []string) ([]*Uni
 				}
 			}
 			if len(missing) > 0 {
-				return nil, fmt.Errorf("cannot get unit %q status: missing %s in ‘systemctl show’ output", cur.UnitName, strings.Join(missing, ", "))
+				return nil, fmt.Errorf("cannot get unit %q status: missing %s in ‘systemctl show’ output", cur.Id, strings.Join(missing, ", "))
 			}
 			sts = append(sts, cur)
+
+			// If we received more replies than what we requested, bail.
 			if len(sts) > len(unitNames) {
-				break // wut
+				break
+			}
+
+			// Record which unit 'Name' request produced this status because
+			// if the request was made using an aliased name, we must be
+			// consistent and reply using the same alias. We will check the
+			// validity of the alias below.
+			cur.Name = unitNames[len(sts)-1]
+
+			// The 'Names' DBUS property of the 'Unit' interface exploses all
+			// the unit name aliases as well as the real name 'Id'. In
+			// order to verify if the request match the reply, we should compare
+			// the request name 'Name' against the 'Names' list for incase the
```

Comment:
maybe "for incase" -> "in case" (at least to my non-native ears this sounds easier :-) )